src/ui/Facet/FacetSearchElement.ts

```diff
@@ -0,0 +1,139 @@
+import { $$ } from '../../utils/Dom';
+import { SVGIcons } from '../../utils/SVGIcons';
+import { SVGDom } from '../../utils/SVGDom';
+import { Component } from '../Base/Component';
+import { l } from '../../strings/Strings';
+import { EventsUtils } from '../../utils/EventsUtils';
+import { PopupUtils, PopupHorizontalAlignment, PopupVerticalAlignment } from '../../utils/PopupUtils';
+
+export class FacetSearchElement {
+  public search: HTMLElement | undefined;
+  public magnifier: HTMLElement | undefined;
+  public wait: HTMLElement | undefined;
+  public clear: HTMLElement | undefined;
+  public input: HTMLInputElement | undefined;
+  public searchBarIsAnimating: boolean = false;
+  public searchResults: HTMLElement | undefined;
+
+  constructor() {
+    this.searchResults = document.createElement('ul');
+    this.searchResults.style.display = 'none';
+    $$(this.searchResults).addClass('coveo-facet-search-results');
+  }
+
+  public build(handleFacetSearchKeyUp: (e: KeyboardEvent) => void, handleFacetSearchClear: () => void, handleFacetSearchFocus: () => void) {
```

Comment:
It's possible to break the icon build/append logic (e.g. magnifier, wait, clear) as well as the input creation/event binding into separate methods to make the `build()` smaller.